### ISSUE TITLE
Not working with Node-Red v1.3.5 - TypeError: Cannot read property 'localeCompare' of undefined

### ISSUE THREAD

**Issue by Gun-neR** [tek79/node-red-contrib-i2clcd-alt#2]
Hello.  I recently came across this while testing many LCD nodes.  This one seemed the simplest (once I figured out how to manually install it :P ) to operate.  However, I only ever get this "difficult for Google to interpret" error

TypeError: Cannot read property 'localeCompare' of undefined

https://discourse.nodered.org/t/trying-to-send-text-to-lcd-over-i2c-contrib/47873/50

**Comment by tek79** [comment]
I'm not sure what you mean when you say "manually install." Sorry, I'm not going to read through that entire thread you've linked. Without knowing more about your hardware, OS, Node.js version, etc., it's going to be a bit difficult to assist.

You should be installing the node with this command:
`npm install tek79/node-red-contrib-i2clcd-alt`

If you are doing it any other way, you are probably missing the dependencies. If you get any complaints during the installation of the node with the above method, you may be missing dev tool dependencies. The command below may or may not be correct for your OS, so please research this first.

`sudo apt-get install -y build-essential python3-dev`

I am currently using this node with Node.js 12 and Node-RED 1.3.5 with no issues on a few different platforms and OS.

**Comment by Gun-neR** [comment]
OS: Raspbian 9 (stretch)
NPM: 7.18.1
Node-Red: 1.3.5
NodeJS: 16.4.0

By "manual" I meant via **get http://...**  as it was not installable from within Node-Red pallets with a single click :) (I am new to NR and was a bit lost how to install it otherwise... but the way I did got the node installed, so I thought I was good to go.)

I reinstalled your contrib again with the command you just supplied... thanks for that... if it was on your site I didn't see it (I see it is there now, thanks!).

I was already up to date on the python2-dev

LCD works perfectly from Python script on RPi, so all hardware, wiring and i2c settings are good.

As are the settings in the node:

```
Name             i2clcd-alt
i2c address      0x3f
i2c bus number   1
LCD columns      20
LCD rows         4
```

But I am still seeing same error coming from the the node:

```
7/1/2021, 10:38:17 PMnode: i2clcd-alt
msg : error
"TypeError: Cannot read property 'localeCompare' of undefined"

7/1/2021, 10:38:17 PMnode: a101dfa6.8642b
msg.payload : string[4]
"init"

7/1/2021, 10:38:20 PMnode: i2clcd-alt
msg : error
"TypeError: Cannot read property 'localeCompare' of undefined"

7/1/2021, 10:38:20 PMnode: a101dfa6.8642b
msg.payload : string[18]
"line1:Hello Gunner"
```

Any other ideas??

**Comment by Gun-neR** [comment]
OK... I just uninstalled and reinstalled to see if there were any warnings... yep... and NOT loaded into NR this way.

Sorry, I am unsure if these are something I can fix or if I am running something too new a version for your contrib??

```
pi@raspberrypi3:~/.node-red $ npm install tek79/node-red-contrib-i2clcd-alt
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated phantomjs-prebuilt@2.1.16: this package is now deprecated
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

added 2 packages, and audited 469 packages in 43s

24 packages are looking for funding
  run `npm fund` for details

11 vulnerabilities (9 low, 2 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
```

**Comment by Gun-neR** [comment]
I found another contrib that does work :)

But if you want me to test any updates in yours, feel free to let me know and I will reinstall it and do what is needed.

**Comment by tek79** [comment]
Yes, as per the readme, this code was tested on Node 8.x through 12.x. Absolutely no testing has been done with 16.x, as I only use LTS versions. If you had a 'normal' installation of Node-RED on your Pi (installed from the official script,) you would have an older version of Node and not have these issues. It sounds like this is not the case for you.

Also, it looks like you're trying to send:
`line1:Hello Gunner`
as your msg.payload, when in fact you should be sending it like this:
`msg.topic:line1`

`msg.payload:Hello Gunner`

So even if the dependencies had all installed, it still wouldn't have worked. I've added an example flow you can import into Node-RED to see how it is supposed to work, if the above still isn't clear. Glad to hear you found something else that did. This project may eventually support Node 16.x, once it reaches LTS status. You can always try updating the dependencies in the package.json to the latest version and try to install again.

**Comment by tek79** [comment]
Sorry, these are actually already the latest dependency version available. i2c-bus is supported up to Node 16.x, but Sleep may only work up to Node 14.x, as it requires the NAN library, which only supports up to Node 14.x.

Sleep came from one of the libraries that I used as a reference when building this and it's really not good to use anyway, so I will eventually replace it or rewrite the code to work without it. It's been about two years since I've really looked at this code. Also, what library did you end up using?

**Comment by tek79** [comment]
If you want to give it a try, you can install 'V2' of this module, which has not been extensively tested, but is working with Node 16.x. You can install it with the following command `npm install tek79/node-red-contrib-i2clcd-alt#v2`

Please see the [readme](https://github.com/tek79/node-red-contrib-i2clcd-alt/tree/v2) for additional information about this node.

**Comment by Gun-neR** [comment]
Thanks for looking into this... dragging you back into old code and all :)

I did install NR following the official process, but saw the note recommending upgrading... I opted for 16, but now see how bleeding edge may not be so good.  But honestly with all the stuff I have been testing, all seemed OK so far.  Even TenserFlow is chugging along OKish on my old RPi3, albeit with lots of hints that I should make changes ('nother topic for 'nother site).

However, yesterday I downgraded to 14.17.2, just in case.

And Thank You for the added info and example about command issuance, all I was going on was the image and the basic reference `line1:data`, etc.

I will try a reinstall and test sometime today.

As for the other code I found (while on Node 14) it is https://flows.nodered.org/node/node-red-contrib-pcf8574-lcd and I had to use the setting for the PCF8574**AT** variant.  It is a bit more setup (JSON) to kick the data out , and I think I will like your method better for across device transmission... but testing (hardware and my abilities) will tell.
